tests/marqo_test.py

```diff
@@ -5,14 +5,37 @@
 """
 import unittest
 from marqo.utils import construct_authorized_url
-
+from marqo.version import __marqo_version__ as py_marqo_support_version
+from marqo.client import Client
+import os
+import time
 
 class MarqoTestCase(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls) -> None:
         local_marqo_settings = {
-            "url": 'http://localhost:8882'
+            "url": os.environ.get("MARQO_URL", 'http://localhost:8882'),
         }
+
+        api_key = os.environ.get("MARQO_API_KEY", None)
+        if (api_key):
+            local_marqo_settings["api_key"] = api_key
+
         cls.client_settings = local_marqo_settings
         cls.authorized_url = cls.client_settings["url"]
+
+        # class property to indicate if test is being run on multi
+        cls.IS_MULTI_INSTANCE = (True if os.environ.get("IS_MULTI_INSTANCE", False) in ["True", "TRUE", "true", True] else False)
+
+        marqo_server_version = Client(**cls.client_settings).get_marqo()["version"]
+        if marqo_server_version != py_marqo_support_version:
+            print(f"WARNING: supported py Marqo version and Marqo versions aren't the same!\n {marqo_server_version} != {py_marqo_support_version}")
+            print(f"MARQO SERVER VERSION -> {marqo_server_version}")
+            print(f"PY-MARQO SUPPORTED VERSION -> {py_marqo_support_version}")
+
+
+
+    def warm_request(f) -> None:
```

Comment:
add  docstring about what this does and when it should be used

Reply:
are you sure this runs the command 5 times? or does it execute once and do some no-op with the result?

Reply:
what might work is something like this:
```
def warm_request(func, *args, **kwargs):
    for i in range(5):
        func(*args, **kwargs) 
```

Reply:
yep the command runs it 5 times based on the logs, but I can modify the code as per your suggestion